### PR TITLE
Don't change cursor to hand for disabled buttons

### PR DESCRIPTION
**Pull request by tsudoko**

### CODE REVIEW

File: ui_buttons.h [hunk 1]
```diff
@@ -12,6 +12,7 @@ static void button_setcolors_success(BUTTON *b)
     b->c3 = COLOR_BUTTON_SUCCESS_HOVER_BACKGROUND;
     b->ct1 = COLOR_BUTTON_SUCCESS_TEXT;
     b->ct2 = COLOR_BUTTON_SUCCESS_HOVER_TEXT;
+    b->disabled = 0;
```
Comment (muchweb) [inline ui_buttons.h]: These functions are meant for colours only (as name suggests). Button colour is a *effect* of being disabled, not an *cause*.

Comment (tsudoko) [inline ui_buttons.h]: Can you think of a better way that doesn't duplicate code too much? I could put `b->disabled = 1`/`b->disabled = 0` outside those functions, but that would only add more unneeded code. Or should I just rename color functions?

**Comment by GrayHatter** [comment]
I agree with @muchweb on this one, that's not the best place to be changing button status...

**Comment by Doom032** [comment]
I tried this but I still see the pointer cursor when I hover over disabled buttons. I rebased it on master to make it compile. Does something need updating for this to work?

**Comment by tsudoko** [comment]
Works here after rebasing.

**Comment by GrayHatter** [comment]
@irungentoo this is ready to be merged

**Pull request merged**